sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegen.scala

```diff
@@ -297,7 +297,7 @@ case class WholeStageCodegen(child: SparkPlan) extends UnaryNode with CodegenSup
     "pipelineTime" -> SQLMetrics.createTimingMetric(sparkContext,
       WholeStageCodegen.PIPELINE_DURATION_METRIC))
 
-  override def doExecute(): RDD[InternalRow] = {
+  private[sql] def doCodeGen(): (CodegenContext, String) = {
```

Comment:
you can remove the private[sql] - everything in execution is considered private.

Can you add comment explaining what the return tuple means?